lib/base-service.js

```diff
@@ -0,0 +1,123 @@
+const CrudBaseService = require('./crud-base-service');
+const mongoose = require('mongoose');
+const Model = require('./model');
+
+class BaseService extends CrudBaseService {
+  constructor() {
+    if (mongoose.connection.readyState !== 1) {
+      throw new Error('mongoose is not connected');
+    }
+    super(Model);
+  }
+
+  static getFieldName(request) {
+    let fieldName = '';
+    Object.keys(request).forEach((key) => {
+      if (key !== 'id') {
+        fieldName = key;
+      }
+    });
+    return fieldName;
+  }
+}
+
+const subDocumentMethods = {
+  'push': function(call) {
+    return this.Model.findOne({'_id': call.request.id})
+      .then((model) => {
+        if (model === null) {
+          CrudBaseService.throwNotFoundError();
+        }
+        const fieldName = BaseService.getFieldName(call.request);
+        call.request[fieldName].forEach((child) => {
+          model[fieldName].push(child);
+        });
+        return model.save();
+      }).then(() => {
+        return {};
+      });
+  },
+  'addToSet': function(call) {
+    return this.Model.findOne({'_id': call.request.id})
+      .then((model) => {
+        if (model === null) {
+          CrudBaseService.throwNotFoundError();
+        }
+        const fieldName = BaseService.getFieldName(call.request);
+        call.request[fieldName].forEach((child) => {
+          child._id = child.id;
+          model[fieldName].addToSet(child);
+        });
+        return model.save();
+      }).then(() => {
+        return {};
+      });
+  },
+  'remove': function(call) {
+    return this.Model.findOne({'_id': call.request.id})
+      .then((model) => {
+        if (model === null) {
+          CrudBaseService.throwNotFoundError();
+        }
+        const fieldName = BaseService.getFieldName(call.request);
+        call.request[fieldName].forEach((child) => {
+          if (!existChild(model[fieldName], child)) {
+            /* eslint-disable no-console */
+            console.warn(`sub document id '${child.id}' is not exist in '${fieldName}'`);
```

Comment:
change "sub document id '${child.id}' is not exist in '${fieldName}' " to "sub document id '${child.id}' does not exist in '${fieldName}'"